core/src/main/java/org/elasticsearch/action/admin/indices/upgrade/post/ShardUpgradeResponse.java

```diff
@@ -32,7 +33,9 @@
  */
 class ShardUpgradeResponse extends BroadcastShardResponse {
 
-    private org.apache.lucene.util.Version version;
+    private org.apache.lucene.util.Version luceneVersion;
```

Comment:
If my understanding is correct, this is the version of the "oldest" segment in the index? If yes, can you document it: I had to think about it for a bit since the elasticsearch version below also points to a lucene version which is the version which has been used to write the index initially.